### PR TITLE
Add dataset card tags to docs

### PR DESCRIPTION
**Pull request by stevhliu**
This PR addresses the missing YAML tags in #222.

### CODE REVIEW

File: docs/hub/datasets-cards.md [hunk 1]
```diff
@@ -1,7 +1,30 @@
...
+configs: List[str]
+extra_gated_fields: Dict
+extra_gated_prompt: str
+```
+
+Use the [Dataset Tagger](https://huggingface.co/spaces/huggingface/datasets-tagging) to help you create the appropriate tags.
+
+The [Dataset Card Creation Guide](https://github.com/huggingface/datasets/blob/main/templates/README_guide.md) gives an overview of what components are found in a dataset card. The process for creating a new dataset card is outlined in the [Create a dataset card](https://huggingface.co/docs/datasets/dataset_card) guide. 
 
 Reading through existing dataset cards, such as the [ELI5 dataset card](https://github.com/huggingface/datasets/blob/main/datasets/eli5/README.md), is a great way to familiarize yourself with the common conventions. There is also an [interactive dataset card builder](https://huggingface.co/datasets/card-creator/) that can guide you through creating your card.
```
[the remark this answers is not in the document]
Comment (stevhliu) [reply docs/hub/datasets-cards.md:30]: Great point! I think we can remove:

- Dataset Card Creation Guide since it is already linked in the Create a dataset card (this guide is also more complete). 
- Interactive dataset card builder was created by a community member, and we don't maintain it.

[the remark this answers is not in the document]
Comment (NimaBoscarino) [reply docs/hub/datasets-cards.md:30]: That sounds great to me!

File: docs/hub/datasets-cards.md [hunk 2]
```diff
@@ -1,7 +1,30 @@
...
+Each dataset may be documented by the `README.md` file in the repository. This file is called a **dataset card**, and the Hugging Face Hub will render its contents on the dataset's main page. To inform users about how to responsibly use the data, it's a good idea to include information about any potential biases within the dataset. Generally, dataset cards help users understand the contents of the dataset and give context for how the dataset should be used. 
 
-The [Dataset Card Creation Guide](https://github.com/huggingface/datasets/blob/main/templates/README_guide.md) gives an overview of what components are found in a dataset card. The process for creating a new dataset card is outlined in the [Create a dataset card](https://huggingface.co/docs/datasets/dataset_card) guide.
+You can also add dataset tags to your card. Dataset tags describe important information about a dataset such as its license, language, and size. More importantly, the tags help users discover a dataset on the Hub. Tags are defined in a YAML section at the top of the `README.md` file. Supported tags include:
```
Comment (julien-c) [inline docs/hub/datasets-cards.md:5]:
```suggestion
You can also add dataset tags to your card. Dataset tags describe important information about a dataset such as its license, language, and size. More importantly, the tags help users discover a dataset on the Hub. Tags are defined in a YAML metadata section at the top of the `README.md` file. Supported tags include:
```

File: docs/hub/datasets-cards.md [hunk 3]
```diff
@@ -1,7 +1,30 @@
...
+
+```yaml
+annotations_creators: List[str]
+language_creators: List[str]
+language: List[str]
```
Comment (julien-c) [inline docs/hub/datasets-cards.md:10]:
```suggestion
language: List[str]
language_details: Optional[str]
```

File: docs/hub/datasets-cards.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (julien-c) [reply docs/hub/datasets-cards.md:30]: Maybe the "interactive dataset card builder" can be linked from inside the other resources (including the "Tagger" space) instead of directly here? WDYT @lhoestq?

(Also it's a nit but for consistency with models I would call it something like "Dataset Metadata Creator" and not Tagger because for models we call it Model metadata, not model tagging)

[the remark this answers is not in the document]
Comment (lhoestq) [reply docs/hub/datasets-cards.md:30]: The interactive dataset card builder can be dropped IMO. It's easier to edit the dataset card on the Hub directly no ?

File: docs/hub/datasets-cards.md [hunk 4]
```diff
@@ -1,7 +1,30 @@
 # Dataset Cards
 
-Each dataset may be documented by the `README.md` file in the repository. This file is called a **dataset card**, and the Hugging Face Hub will render its contents on the dataset's main page. To inform users about how to responsibly use the data, it's a good idea to include information about any potential biases within the dataset. Generally, dataset cards help users understand the contents of the dataset and give context for how the dataset should be used.
+Each dataset may be documented by the `README.md` file in the repository. This file is called a **dataset card**, and the Hugging Face Hub will render its contents on the dataset's main page. To inform users about how to responsibly use the data, it's a good idea to include information about any potential biases within the dataset. Generally, dataset cards help users understand the contents of the dataset and give context for how the dataset should be used.
```
Comment (osanseviero) [inline docs/hub/datasets-cards.md:3]: nit
```suggestion
Each dataset may be documented by the `README.md` file in the repository. This file is called a **dataset card**, and the Hugging Face Hub will render its contents on the dataset's main page. To inform users about how to responsibly use the data, it's a good idea to include information about any potential biases within the dataset. Generally, dataset cards help users understand the contents of the dataset and give context for how the dataset should be used.
```

File: docs/hub/datasets-cards.md (hunk 2 above)
Comment (osanseviero) [inline docs/hub/datasets-cards.md:5]:
```suggestion
You can also add dataset metadata to your card. The metadata describes important information about a dataset such as its license, language, and size. It also contains tags that help users discover a dataset on the Hub. Tags are defined in a YAML section at the top of the `README.md` file. Supported tags include:
```

File: docs/hub/datasets-cards.md (hunk 3 above)
Comment (osanseviero) [reply docs/hub/datasets-cards.md:10]: I think some reference to language_detials would be useful

File: docs/hub/datasets-cards.md [hunk 5]
```diff
@@ -1,7 +1,30 @@
...
-The [Dataset Card Creation Guide](https://github.com/huggingface/datasets/blob/main/templates/README_guide.md) gives an overview of what components are found in a dataset card. The process for creating a new dataset card is outlined in the [Create a dataset card](https://huggingface.co/docs/datasets/dataset_card) guide.
+You can also add dataset tags to your card. Dataset tags describe important information about a dataset such as its license, language, and size. More importantly, the tags help users discover a dataset on the Hub. Tags are defined in a YAML section at the top of the `README.md` file. Supported tags include:
+
+```yaml
+annotations_creators: List[str]
+language_creators: List[str]
+language: List[str]
+license: Union[str, List[str]]
+license_details: str
+multilinguality: List[str]
```
Comment (osanseviero) [inline docs/hub/datasets-cards.md:13]: What is this tag for?

**Review by osanseviero — COMMENTED** [review]
I just left some minor comments

File: docs/hub/datasets-cards.md (hunk 5 above)
Comment (lhoestq) [reply docs/hub/datasets-cards.md:13]: Possible values are
```
{
  "monolingual": "contains a single language",
  "multilingual": "contains multiple languages",
  "translation": "contains translated or aligned text",
  "other": "other type of language distribution"
}
```
It's not particularly useful IMO, maybe at one point we can just drop it

Comment (osanseviero) [reply docs/hub/datasets-cards.md:13]: GIven we have `language` I don't feel it's super useful. The first 2 can be obtained from `language`. `translation` sounds more like a tag for me.

**Review by NimaBoscarino — APPROVED** [review]

**Review by osanseviero — APPROVED** [review]
Thanks for this PR! Very cool!

FYI, if you would like a second pass review, you can re-request a review in the "Reviewers" panel at top right, that way we know if you're expecting an action from us or not :D

File: docs/hub/datasets-cards.md [hunk 6]
```diff
@@ -2,6 +2,27 @@
...
+pretty_name: str
+size_categories: List[str]
+source_datasets: List[str]
+task_categories: List[str]
+task_ids: List[str]
+paperswithcode_id: str
+train-eval-index: List[Dict]
+configs: List[str]
+extra_gated_fields: Dict
+extra_gated_prompt: str
```
Comment (lhoestq) [inline docs/hub/datasets-cards.md:23]: to help users filter datasets by keywords, they can use `tags`:
```suggestion
extra_gated_prompt: str
tags: List[str]
```

File: docs/hub/datasets-cards.md (hunk 5 above)
Comment (julien-c) [reply docs/hub/datasets-cards.md]: Also it's NLP-specific

I would also advocate to drop it going forward

File: docs/hub/datasets-cards.md (hunk 6 above)
Comment (julien-c) [reply docs/hub/datasets-cards.md:23]: yes, can't believe i missed that one!

Comment (osanseviero) [reply docs/hub/datasets-cards.md:23]: I would actually put `tags` as one of the first fields
